src/background.js

```diff
@@ -137,8 +137,13 @@ const stateManager = {
     // In other words, if the user has made their own decision for DoH,
     // then we want to respect that and never run the heuristics again
 
-    // On Mismatch - run never run again (make init check a function)
+    // If there was no previous TRR.MODE recorded (-1) or the current and previous modes match,
+    // we should run heuristics!
+    if (prevMode === -1 || prevMode === curMode){
```

Comment:
Is `prevMode === -1` sufficient here? Do we need `prevMode === -1 && curMode === 0`?

Reply:
I don't think so. We only care if there's a mismatch of previous and current mode at this stage. We have other logic to process the current trr.mode before hand, I think.

Reply:
Confirming now.

Reply:
Confirmed. This is not required. It only time `prevMode` is `-1` is on first run.

Reply:
What if prevMode === -1 and curMode === 2? (i.e. user-set for some reason. or it's a stranded user and for some reason our previous TRR mode pref is gone)

Reply:
adding `&& curMode === 0` replicates the logic before, when prevMode was (wrongly) 0. I think this is safer.

Reply:
They're detected during the [`trrModePrefHasUserValue`](https://github.com/mozilla/doh-rollout/blob/master/src/background.js#L314-L336).